### PR TITLE
Only show reviews for listed add-ons (bug 1208408)

### PR DESCRIPTION
**Pull request by kumar303**

### CODE REVIEW

**Comment by tofumatt** [comment]
r+

**Comment by kumar303** [comment]
Thanks @tofumatt 

I should mention that I also got confirmation from Jorge that limiting reviews to listed add-ons is fine. Their expectation was that only listed add-ons should ever have reviews so--as mentioned in the bug comments--this is just to fix rare edge cases.

File: apps/editors/tests/test_views.py [hunk 1]
```diff
@@ -1245,6 +1245,16 @@ def test_no_reviews(self):
         eq_(doc('.no-results').length, 1)
         eq_(doc('.review-saved button').length, 1)  # Show only one button.
 
+    def test_do_not_show_reviews_for_unlisted_addons(self):
+        Addon.objects.all().update(status=amo.STATUS_NULL)
```
Comment (magopian) [inline apps/editors/tests/test_views.py]: Addons with `STATUS_NULL` aren't unlisted, but incomplete.

To have an Addon unlisted, you need to update its `is_listed` field to `False`:
`Addon.objects.all().update(is_listed=False)`

As a side note, once they're unlisted, they won't be visible using the `Addon.objects` manager, you'll have to use either `Addon.with_unlisted` or `Addon.unfiltered` (which also includes deleted addons, which have `STATUS_DELETED`).

File: apps/editors/tests/test_views.py [hunk 2]
```diff
@@ -1245,6 +1245,16 @@ def test_no_reviews(self):
...
+
+        r = self.client.get(self.url)
```
Comment (magopian) [inline apps/editors/tests/test_views.py]: We try to not use single letter variables

File: apps/editors/tests/test_views.py [hunk 3]
```diff
@@ -1245,6 +1245,16 @@ def test_no_reviews(self):
...
+        eq_(r.status_code, 200)
```
Comment (magopian) [inline apps/editors/tests/test_views.py]: We switched to using pytest, so when possible, we use simple `assert`s

File: apps/editors/views.py [hunk 4]
```diff
@@ -482,7 +482,12 @@ def queue_fast_track(request):
 def queue_moderated(request):
     rf = (Review.objects.exclude(Q(addon__isnull=True) |
                                  Q(reviewflag__isnull=True))
-                        .filter(editorreview=1)
+                        .filter(editorreview=1,
+                                # Only show reviews for listed add-ons.
+                                # Unlisted add-ons typically won't have
+                                # reviews but they might if their status
+                                # is ever regressed.
+                                addon__status__in=amo.LISTED_STATUSES)
```
Comment (magopian) [inline apps/editors/views.py]: There's (sadly) some confusion in the code between an add-on being [listed (and not deleted, pending, incomplete...)](https://github.com/mozilla/olympia/blob/48911a1ed3f04b81033206f8b040c573573a859f/apps/constants/base.py#L101) and being [listed (and not unlisted)](https://github.com/mozilla/olympia/blob/48911a1ed3f04b81033206f8b040c573573a859f/apps/addons/models.py#L361)

**Comment by magopian** [comment]
I think the changes here make sense, but are missing the case where an add-on is unlisted (`is_listed=False`).
